### ISSUE TITLE
EGD: encoding mixed-metre verse

### ISSUE THREAD

**Issue by danbalogh** [erc-dharma/project-documentation#192]
@arlogriffiths, I'm starting to move some topics raised in the EGD to issues here, for attention at an opportune time. None of these are urgent, but perhaps this way the EGD will be less cluttered and more responsive, and these issues will not be lost from view. Here is the first one.

We have not provided explicitly for the kind of upajāti that mixes 11-syllable and 12-syllable lines (triṣṭubh and jagatī types). I don't think I've ever seen an epigraphic example before now, but this sort of stanza is pretty common in the MBh. I have one now where the first line is in the vaṁśastha pattern, while the next three are in upendravajrā. (Vengi Calukya corpus 00070, Unpublished Eastern Cālukya CP, late 10th century.)

For the time being, I'm encoding this as upendravajrā, with `@real` on the anomalous first quarter. But this sort of implies that the line is metrically faulty, which is only true for classical Sanskrit verse, but it's a legitimate form in a looser set of conventions.
We could instead:
A. encode it as Vedic trimeter - but the rules there are much more lax, and the pattern is trochaic, so this is also inaccurate;
B. introduce yet another subrule for encoding metre (e.g. add `@met="vaṁśastha"` for the first line but keep "upendravajrā" or "upajāti" for the stanza as a whole), but this would complicate our approach further for the sake of a very uncommon phenomenon.
C. [I'm adding this now, as a complex but perhaps most satisfactory option that would cover a wide range of cases]: introduce a metre="mixed" option that could only be used on `<lg>` elements, and if a stanza has mixed metre, then `@met` would be mandatory on all of its lines.

In case you are interested, the stanza is
yaśo-vikāśāya vikāśitā diśo
daśāpi viśvasya pitāmahena|
śriyan nidhā(?yo)rasi yasya viṣṇu+ḥ
sva-mūrttitāṁ svaṁ ca tapāṁsi tepe||
I don't really understand the second half; the reading is very clear but emendation may be necessary. It may be a Puranic quote (though Google has not helped find it); pls let me know if it rings a bell.

**Comment by arlogriffiths** [comment]
I am inclined to favor option C. I cannot oversee the magnitude of the potential ripple effect of stanzas already encoded that will have to be reencoded should that rule be introduced. A way to find out might be to ask Axelle to generate a list of all stanzas in which @real has been used on any line.

**Comment by danbalogh** [comment]
@arlogriffiths , I assume that all instances where real has been used will be actual metrical anomalies and not cases of mixed metres, since nobody has raised that problem before. But I don't mind looking through such a list as a precaution.

@ajaniak , could you generate that list for me? Ideally, it would be lovely to have the text right there (i.e. each item on the list would be the complete contents of an `<lg>` that contains an `<l>` with `@real` on it), but if that is difficult to produce, then a list of filenames containing `<l>` with `@real` will also be OK; I'll then look up the texts manually.

**Comment by danbalogh** [comment]
I have checked through the list generated by Axelle, looking at many of them directly in the original XML, though after a while I just skimmed through the dozens of instances in the Khmer corpus, all of which concern vipulā (or strange) anuṣṭubh stanzas. The point is: there are no stanzas anywhere in the corpus that we need to change retrospectively if we introduce "mixed" as suggested above.

I have made the suggested changes (plus some more details) in the EGD, §§2.3.4 and 2.3.5.

@ajaniak: I am not sure to what extent our schema constrains the use of `@met`; I have encoded my mixed-metre stanza as suggested here, and it still validates, so it may not be necessary to do anything to the schema. But if you agree, I think it would be nice to introduce the following:
if an `<lg>` has `@met` with the value "mixed", then every `<l>` that is a descendant of that `<lg>` must have _either_ `@met` _or_ `@real` (or both)
